core/src/main/java/org/jivesoftware/spark/ui/login/CertificatesManagerSettingsPanel.java

```diff
@@ -152,7 +157,33 @@ private void showCertificate() {
 		CertificateDialog certDialog = new CertificateDialog(localPreferences,
 				certControll.getCertificates().get(certTable.getSelectedRow()));
 	}
-
+	
+	private void addCertificate(){
+		
+		fileChooser.setAcceptAllFileFilterUsed(false);
+		fileChooser.addChoosableFileFilter(certFilter);
+		fileChooser.setFileFilter(certFilter);
+
+		int retVal = fileChooser.showOpenDialog(this);
+		if (retVal == JFileChooser.APPROVE_OPTION) {
+
+			File file = fileChooser.getSelectedFile();
+			
+			try {
+				fileField.setText(file.getAbsolutePath());
+				String certAlias = (String) JOptionPane.showInputDialog(null, Res.getString("dialog.provide.alias"));
+				if (certAlias == null) {
+					throw new IllegalArgumentException();
+				}
+				certControll.addCertificateToKeystore(file, certAlias);
+			} catch (KeyStoreException | CertificateException | NoSuchAlgorithmException | IOException ex) {
+				Log.error("Cannot upload certificate file", ex);
+			} catch (IllegalArgumentException ex) {
+				Log.warning("Certificate alias cannot be null");
```

Comment:
Please, always log the exception if you have one. Instead of `Log.warning("blah")` do `Log.warning("blah",ex)`